src/v2/repositories/implementations/xcm/PendulumXcmRepository.ts

```diff
@@ -0,0 +1,96 @@
+import { BN } from '@polkadot/util';
+import { XcmTokenInformation } from 'src/modules/xcm';
+import { container } from 'src/v2/common';
+import { ExtrinsicPayload, IApi, IApiFactory } from 'src/v2/integration';
+import { Asset } from 'src/v2/models';
+import { Symbols } from 'src/v2/symbols';
+import { XcmRepository } from '../XcmRepository';
+import { XcmChain } from 'src/v2/models/XcmModels';
+import { TokensAccounts } from 'src/v2/repositories/implementations/xcm/AcalaXcmRepository';
+import { decodeAddress } from '@polkadot/util-crypto';
+
+/**
+ * Used to transfer assets from Pendulum
+ */
+
+const PEN = 'Native';
+
+export class PendulumXcmRepository extends XcmRepository {
+  constructor() {
+    const defaultApi = container.get<IApi>(Symbols.DefaultApi);
+    const apiFactory = container.get<IApiFactory>(Symbols.ApiFactory);
+    const registeredTokens = container.get<XcmTokenInformation[]>(Symbols.RegisteredTokens);
+    super(defaultApi, apiFactory, registeredTokens);
+  }
+
+  public async getTransferCall(
+    from: XcmChain,
+    to: XcmChain,
+    recipientAddress: string,
+    token: Asset,
+    amount: BN,
+    endpoint: string
+  ): Promise<ExtrinsicPayload> {
+    if (!to.parachainId) {
+      throw `Parachain id for ${to.name} is not defined`;
+    }
+    if (token.id !== '18446744073709551634') {
+      throw 'Token must be PEN';
+    }
+    let tokenData = PEN;
```

Comment:
Then we would probably need to change this to `let tokenData = token.originAssetId === "PEN" ? "Native" : { XCM: token.originAssetId }` or similiar. Not sure what exactly `token.originAssetId` can be.